preallocator.py

```diff
@@ -29,7 +31,7 @@ def __init__(self, vmms):
     def poolSize(self, vmName):
         """ poolSize - returns the size of the vmName pool, for external callers
         """
-        if vmName not in self.machines.keys():
+        if vmName not in list(self.machines.keys()):
```

Comment:
same comment with regards to the key situation in jobqueue

Reply:
self.machines here is also either a TangoNativeDictionary or TangoRemoteDictionary

Reply:
refer to comment below